src/PrestaShopBundle/Security/Admin/EmployeeProvider.php

```diff
@@ -58,12 +58,21 @@ public function __construct(LegacyContext $context)
      */
     public function loadUserByUsername($username)
     {
-        if (isset($this->legacyContext->employee) && $this->legacyContext->employee->email == $username) {
-            $employee = new Employee($this->legacyContext->employee);
-            $employee->setRoles(
-                Access::getRoles($this->legacyContext->employee->id_profile)
-            );
-            return $employee;
+        static $employees = array();
```

Comment:
And what about a class property to store the cache (which can be reset with another function)? I remember the issues you got with @tomlev :)

Reply:
nope, there is better: using the cache system from Symfony. But I don't know yet if it's the right time to introduce it as I don't have time to introduce it almost everywhere + document it .

Reply:
as soon as my build is green, I'll try to introduce Symfony cache for this class *only* in a specific commit.